src/postgres/getClientConfig.ts

```diff
@@ -5,41 +5,47 @@
 
 import { Client, ClientConfig } from "pg";
 import { ConnectionOptions } from "tls";
+import { postgresDefaultPort } from "../constants";
 import { localize } from "../utils/localize";
 import { nonNullProp } from "../utils/nonNull";
 import { invalidCredentialsErrorType } from "./tree/PostgresDatabaseTreeItem";
 import { PostgresServerTreeItem } from "./tree/PostgresServerTreeItem";
 
 export async function getClientConfig(treeItem: PostgresServerTreeItem, databaseName: string): Promise<ClientConfig> {
-    const { username, password } = await treeItem.getCredentials();
-
+    let { username, password } = await treeItem.getCredentials();
+    if (!(username && password)) {
+        username = treeItem.connectionString.username;
+        password = treeItem.connectionString.password;
+    }
+    let host: string;
+    const ssl: ConnectionOptions = {
+        // Always provide the certificate since it is accepted even when SSL is disabled
+        // Certificate source: https://aka.ms/AA7wnvl
+        ca: BaltimoreCyberTrustRoot
+    };
+    let clientConfig: ClientConfig;
```

Comment:
nit: No need to define host, clientConfig, and client up here with `let`. Just define them as a `const` below